boot/seal.go

```diff
@@ -832,16 +867,19 @@ func buildBootAssets(bootFiles []bootloader.BootFile, modeenv *Modeenv) (assets
 
 	// the last element is the kernel which is not a boot asset
 	for i, bf := range bootFiles[:len(bootFiles)-1] {
-		name := filepath.Base(bf.Path)
+		path := bf.Path
+		name, ok := trustedAssets[path]
+		if !ok {
+			return nil, kernel, nil
```

Comment:
this probably merit a comment why the new logic is ok with not finding some asset chains

Reply:
This one can only be generated by a broken bootloader. So it should be an error.